index.bs

```diff
@@ -52,72 +49,205 @@ RTCPeerConnection, canvas or MediaRecorder.
+### Implicit signaling
+A common use case for processors and generators is to connect the media flow from
+a pre-existing platform track (e.g., camera, screen capture) to pre-existing platform sinks
+(e.g., peer connection, media element) with a transformation in between, in a chain like this:
+
+Platform Track -> Processor -> Transform -> Generator -> Platform Sinks
+
+Absent the Breakout Box elements, the platform sinks would normally send the signals directly
+to the platform track (and its source). Arguably, in many cases the source of the platform track
+can still be considered the source for the platform sinks and it would be desirable to keep
+the original platform signaling even with the Processor -> Transform -> Generator chain between
+them. This can be achieved by assigning the platform track to the
+{{MediaStreamTrackGeneratorInit/signalTarget}} field in
+{{MediaStreamTrackGeneratorInit}}. Note that such a connection between the platform track and the
+generator includes all possible internal signaling coming from the platform sinks and not just
+the generic signals exposed as {{MediaStreamTrackSignal}} objects via the
+{{MediaStreamTrackGenerator/readableControl}} and
+{{MediaStreamTrackProcessor/writableControl}} fields. Just connecting explicit signals from
+a generator to a processor (e.g., with a call like
+`generator.readableControl.pipeTo(processor.writableControl`) forwards only
+the Web Platform-exposed signals, but ignores any internal custom signals as there is no
+way for the platform sinks to know what the upstream source is.
 
-All signals are ignorable; the downstream objects will do whatever makes sense with frames
-sent on the stream, no matter what the signals indicate.
 
-The "stop" and "mute" signals both indicate that frames should no longer be sent. "stop" indicates that the
-MediaStreamTrack has stopped, and that no frames will ever be required again.
+# Examples # {#examples}
+TODO
 
-THe "configure" signal has semantics equivalent to ApplyConstraints on the MediaStreamTrack, with all
-constraints set to "ideal": It indicates that more efficient downstream processing will be achieved if
-subsequently generated frames conform to the size and rate constrains given.
+# Security considerations # {#security-considerations}
 
-# Examples # {#examples}
+The security of this API relies on existing mechanisms in the Web platform.
+For example, data from cross-origin resources cannot be accessed due to existing restrictions
+to access such resources (e.g., it is not possible to access the pixels of a cross-origin
+image or video element). Data from cameras, microphones or the screen is subject to user
+authorization.
+The media data this API exposes is already available through other APIs (e.g., media elements +
+canvas + canvas capture). In addition to the media data, this API exposes some control signals
+such as requests for new frames. These signals are intended as hints and do not pose a
+significant security risk.
```

Comment:
Should we mention expliicitly that we follow the same rules on origin-tainted buffers as WebCodecs? (Not sure that we actually support those anywhere, but it's one of the questions people always ask when you have a data-processing interface - "what about tainted data").
I think Canvas is said to have the most words about those things.

Reply:
Gave it a try. Please take another look.